evm/utils/headers.py

```diff
@@ -65,3 +69,29 @@ def compute_gas_limit(parent_header, gas_limit_floor):
         return parent_header.gas_limit + decay
     else:
         return gas_limit
+
+
+def generate_header_from_parent_header(
+        compute_difficulty,
```

Comment:
naming nitpick.  Might be a bit clearer that this is a callable if it was named `compute_difficulty_fn` or something else with a clear indicator that it's a callable.